sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,482 @@ def reduce_inequalities(inequalities, symbols=[]):
+
+    return A, B, C, standard_constraints
+
+def linprog_maximize_from_equations(constraints, objective, variables):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method. All coefficients must be Rational or
+    floats.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        Stict inequalities (>, <) and not equals are not allowed.
+
+    objective : a linear function to maximize
+        If you want to minimize a function, f, simply make the objective
+        function -f.
+
+    variables : a list of variables included in constraints and objective
+
+    Returns
+    =======
+
+    Returns a tuple of solutions optimum, argmax:
+
+    optimum : Rational or Float
+
+    argmax : dictionary of variables to Rationals or Floats
+
+    argmax_dual : dictionary of constraints to Rationals or Floats
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linprog_maximize_from_equations
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual  = linprog_maximize_from_equations([r1,r2,r3], x+y+5*z, [x, y, z])
+    >>> optimum
+    11/3
+    >>> argmax
+    {x: 0, y: 1/3, z: 2/3}
+    >>> argmax_dual
+    {-x - 3*z <= -2: 2/3, y + 2*z <= 3: 0, 2*x + y + 7*z <= 5: 1}
+
+    See Also
+    ========
+
+    solve_univariate_inequality
+    simplex
+    find_feasible
+    """
+    A, B, C, standard_constraints = _linear_programming_to_matrix(constraints, objective, variables)
+    optimum, argmax, argmax_dual = _simplex(A, B, C)
+    argmax = {variables[i] : argmax[i] for i in range(len(variables))}
+    argmax_dual = {standard_constraints[i] : argmax_dual[i] for i in range(len(standard_constraints))}
+    return optimum, argmax, argmax_dual
+
+
+def find_feasible(constraints, variables):
+    """
+    Finds a feasible solution to a system of linear inequalities, if any exist,
+    with the first phase of the simplex method.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities and equalities
+        Stict inequalities (>, <) and not equals are not allowed.
+
+    variables : list of variables included in inequalities
+
+    Returns
+    =======
+
+    dictionary or None
+        If no feasible solutions exist, returns None. Otherwise, returns a dictionary
+        of variable to values that satisfies the constraints.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import find_feasible
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> find_feasible([r1, r2, r3], [x, y, z])
+    {x: 2, y: 0, z: 0}
+    >>> r1 = x <= 3
+    >>> r2 = x >= 4
+    >>> find_feasible([r1, r2], [x]) is None
+    True
+
+    See Also
+    ========
+
+    linprog_maximize
```

Comment:
`linprog_maximize_from_equations`

Reply:
I think it would be nice to have the general "front door" for these functions be `optimize(a, b, c)` where the correct call is determined by the type of `b`: if a list, use matrix call, otherwise the equation call. (I have never really liked "linear_programming" as the name for this procedure though I understand the historical use of the phrase is meaningful for those who work in this area. The term "optimize" is more likely to catch the interest of someone that perhaps is less familiar with the more technical name.)

Reply:
Does SymPy have other functions that also do some other form of optimization? Is it likely that such functions will be implemented? If so, maybe `optimize` could be a bit confusing as it's more general. 

I don't think it's a good idea to combine `linprog_maximize_from_equations` and `linprog_maximize_from_equations`  into one.  They have different return types, and the inputs are structured very differently. Under the maximization interpretation, matrix `A`, and column vector `B` represent the `constraints` while row vector `C` corresponds to `objective`.  `linprog_maximize_from_equations` has `variables` which there's no equivalent of in `linprog_maximize_from_equations.`

Reply:
I am not suggesting that the code be merged together -- and maybe it's not even a good idea to have a program to let either be called by the type of input (i.e. don't have a polymorphic signature).  But `linprog_from_matrices`  and `linprog_maximize_from_equations` both return `(optimum, argmax, and argmax_dual)` so they are just starting the same problem from different points, aren't they? 

Perhaps we can get rid of the `maximize` from the `equations` name and then have a general "front-door" like the following (which for most users will be the most friendly way to call the solver):

```python
def linprog(eq, constrations):
    eq = sympify(eq)
    constraints = Tuple(*constraints)
    x= eq.free_symbols
    assert x == constraints.free_symbols  # is this a correct assumption?
    return linprog_from_equations(constraints, eq, x)
```
I'll have to look and see how the variables are handled in the return value. It would make most sense if they are returned in a dictionary so one doesn't have to worry about ordering of `x`.

Reply:
> is this a correct assumption?

It might not be. You could have some variables in the constraints that don't appear in the objective function. These variables could still affect the objective by association with variables in the objective. I guess you could also have some variables in the objective function that aren't in the constraints.

> have a "front-door" like the following

After thinking about it more, the return types of the input and output don't have to be a problem. You could combine A and B into one matrix and pass that in.

For example, these constraints
x  +    y < 3
2x + 2y < 1
could be represented like this

```
>>> A = Matrix([[1,1],[2, 2]])
>>> B = Matrix([[3],[1]])
>>> AB = Matrix([[A,B]])
>>> AB
Matrix([[1, 1, 3],
        [2, 2, 1]])
>>> C = Matrix([[1, 2]]) # for objective function to maximize: x + 2y
>>> linprog(C, AB)
```

> Perhaps we can get rid of `maximize`

`linprog`, the function you're suggesting, is doing maximization. Combining A and B into AB and passing that in as constraints is framing things as maximization. A and B don't necessarily represent constraints. In the dual minimization problem, B is the objective function and A and C represent the constraints the dual variables are under.  

`linprog_maximize_from_equations` is named that way because it takes the maximization interpretation. You could make a similar minimization function for the standard minimization problem but I think that would be less intuitive for users—or at least the dual minimization problem is less intuitive to me.

Of course, if a user want to minimize a function, f, under some constraints, you can still use `linprog_maximize_from_equations`. You just need to pass in -f to `linprog_maximize_from_equations`.

Reply:
OK, I'm not sure than any further changes need to happen from my perspective.